javalib/src/main/scala/java/util/Collections.scala

```diff
@@ -0,0 +1,1194 @@
+      })
+  }
+
+  final lazy val EMPTY_LIST: List[_] = {
+    unmodifiableList(
+      new AbstractList[Any] with Serializable with RandomAccess {
+        override def get(index: Int): Any =
+          throw new IndexOutOfBoundsException(index.toString)
+
+        override def size(): Int = 0
+      })
+  }
+
+  final lazy val EMPTY_MAP: Map[_, _] = {
+    unmodifiableMap[Any, Any](
+      new AbstractMap[Any, Any] with Serializable {
+        override def entrySet(): Set[Map.Entry[Any, Any]] =
+          EMPTY_SET.asInstanceOf[Set[Map.Entry[Any, Any]]]
+      })
+  }
+
+  private lazy val EMPTY_ITERATOR: Iterator[_] =
+    new EmptyIterator
+
+  private lazy val EMPTY_LIST_ITERATOR: ListIterator[_] =
+    new EmptyListIterator
+
+  private lazy val EMPTY_ENUMERATION: Enumeration[_] = {
+    new Enumeration[Any] {
+      def hasMoreElements: Boolean = false
+
+      def nextElement(): Any =
+        throw new NoSuchElementException
+    }
+  }
+
+  // Differs from original type definition, original: [T <: jl.Comparable[_ >: T]]
+  def sort[T <: jl.Comparable[T]](list: List[T]): Unit =
+    sort(list, naturalComparator[T])
+
+  def sort[T](list: List[T], c: Comparator[_ >: T]): Unit = {
+    val sortedList = list.sorted(c)
+    list match {
+      case list: RandomAccess => copyImpl(sortedList.iterator, list)
+      case _                  => copyImpl(sortedList.iterator, list.listIterator)
+    }
+  }
+
+  def binarySearch[T](list: List[_ <: jl.Comparable[_ >: T]], key: T): Int =
+    binarySearchImpl(list, (elem: Comparable[_ >: T]) => elem.compareTo(key))
+
+  def binarySearch[T](list: List[_ <: T], key: T, c: Comparator[_ >: T]): Int =
+    binarySearchImpl(list, (elem: T) => c.compare(elem, key))
+
+  def reverse(list: List[_]): Unit =
+    reverseImpl(list)
+
+  def shuffle(list: List[_]): Unit =
+    shuffle(list, new Random)
+
+  def shuffle(list: List[_], rnd: Random): Unit =
+    shuffleImpl(list, rnd)
+
+  def swap(list: List[_], i: Int, j: Int): Unit =
+    swapImpl(list, i, j)
+
+  def fill[T](list: List[_ >: T], obj: T): Unit = {
+    list match {
+      case list: RandomAccess =>
+        (0 until list.size).foreach(list(_) = obj)
+
+      case _                  =>
+        val iter = list.listIterator
+        while (iter.hasNext) {
+          iter.next()
+          iter.set(obj)
+        }
+    }
+  }
+
+  def copy[T](dest: List[_ >: T], src: List[_ <: T]): Unit = {
+    (dest, src) match {
+      case (dest: RandomAccess, src: RandomAccess) => copyImpl(src, dest)
+      case (dest: RandomAccess, _)                 => copyImpl(src.iterator, dest)
+      case (_, src: RandomAccess)                  => copyImpl(src, dest.listIterator)
+      case (_, _)                                  => copyImpl(src.iterator, dest.listIterator)
+    }
+  }
+
+  // Differs from original type definition, original: [T <: jl.Comparable[_ >: T]]
+  def min[T <: AnyRef with jl.Comparable[T]](coll: Collection[_ <: T]): T =
```

Comment:
The original declaration is:
```java
public static <T extends Object & Comparable<? super T>> T min(Collection<? extends T> coll)
```
Im not sure why they put it there in the first place.